abel/rbasex.py

```diff
@@ -0,0 +1,751 @@
+#
+# rBasex — Abel transform for velocity-map images (~spherical harmonics),
+#          similar to pBasex, but with analytical basis functions and without
+#          pixel-grid transformation; a simplified version of the method
+#          described in M. Ryazanov Ph.D. dissertation.
+#
+# pBasex:
+#   Gustavo A. Garcia, Laurent Nahon, Ivan Powis,
+#   “Two-dimensional charged particle image inversion using a polar basis
+#    function expansion”,
+#   Review of Scientific Instruments 75, 4989 (2004).
+#   http://dx.doi.org/10.1063/1.1807578
+#
+# Ryazanov:
+#   Mikhail Ryazanov,
+#   “Development and implementation of methods for sliced velocity map imaging.
+#    Studies of overtone-induced dissociation and isomerization dynamics of
+#    hydroxymethyl radical (CH₂OH and CD₂OH)”,
+#   Ph.D. dissertation, University of Southern California, 2012.
+#   https://search.proquest.com/docview/1289069738
+#   http://digitallibrary.usc.edu/cdm/ref/collection/p15799coll3/id/112619
+#
+###############################################################################
+
+
+# Caches and their parameters
+_prm = None  # [shape, origin, rmax, order, odd]
+_weights = None  # weights — pixel weights
+_dst = None  # Distributions object
+_bs_prm = None  # [Rmax, order, odd]
+_bs = None  # [P[n]] — projected functions
+_ibs = None  # [rbin, wl, wu, cos^n] — arrays for image construction
+_trf = None  # [Af[n]] — forward transform matrices
+_tri_full = None  # [Ai[n]] — inverse-transform matrices without mask and reg
+_tri_prm = None  # [reg] — regularization parameters
+_tri = None  # [Ai[n]] — inverse-transform matrices (or Af for reg='pos')
+
+
+def rbasex_transform(IM, origin='center', rmax='MIN', order=2, odd=False,
+                     weights=None, direction='inverse', reg=None, out='same',
+                     basis_dir=None, verbose=False):
+    r"""
+    This function takes the input image and outputs its forward or inverse Abel
+    transform as an image and its radial distributions.
+
+    The **origin**, **rmax**, **order**, **odd** and **weights** parameters are
+    passed to :class:`abel.tools.vmi.Distributions`, so see its documentation
+    for their detailed descriptions.
+
+    Parameters
+    ----------
+    IM : m × n numpy array
+        the image to be transformed
+    origin : tuple of int or str
+        image origin, explicit in the (row, column) format, or as a location
+        string
+    rmax : int or string
+        largest radius to include in the transform
+    order : int
+        highest angular order present in the data, ≥ 0
+    odd : bool
+        include odd angular orders (enabled automatically if **order** is odd)
+    weights : m × n numpy array, optional
+        weighting factors for each pixel. The array shape must match the image
+        shape. Parts of the image can be excluded from analysis by assigning
+        zero weights to their pixels.
```

Comment:
"The default is None, which applies equal weight to all pixels."

Reply:
Done.